docs/Z-Score-normalization-script.md

```diff
@@ -5,31 +5,54 @@ For some data types, when uploading to cBioPortal, it is currently necessary to
- 
-### How to proceed
-cBioPortal expects z-score normalization to take place per gene. You can produce this extra z-score file yourself, or let cBioPortal do it for your input files by using `convertExpressionZscores.pl`. Most information that now follows was taken from the comments in `convertExpressionZscores.pl`. We've also added an example of the calculation and example of running the program below.
+
 
 ## The cBioPortal convertExpressionZscores method
 Given expression and Copy Number Variation data for a set of samples (patients), generate normalized expression values.
 
-Each **gene** is normalized separately. First, the expression distribution for unaltered copies of the gene is estimated by calculating the mean and variance of the expression values for samples in which the gene is diploid (i.e. value is "0" as reported by [discrete CNA data](File-Formats.md#discrete-copy-number-data)). We call this the unaltered distribution. If the gene has no diploid samples, then its normalized expression is reported as NA. Otherwise, for every sample, the gene's normalized expression is reported as
+Each **gene** is normalized separately. 
+
+We currently have two profiles for generating z-scores using two different base population:
+
+- Distribution based on **diploid** samples only: The expression distribution for unaltered copies of the gene is estimated by calculating the mean and variance of the expression values for samples in which the gene is diploid (i.e. value is "0" as reported by [discrete CNA data](File-Formats.md#discrete-copy-number-data)). We call this the unaltered distribution. If the gene has no diploid samples, then its normalized expression is reported as NA. 
+
+- Distribution based on **all** samples: The expression distribution of the gene
+is estimated by calculating the mean and variance of the expression values for samples whose values are not Null, NA or NaN. If the gene has samples whose expression values are Null, NaN or 'NA', then its normalized expression is reported as NA.
+
+Otherwise when valid expression values exist, with both profiles, for every sample, the gene's normalized expression is reported as
 
 ```
 (r - mu)/sigma
 ```
 
-where `r` is the raw expression value, and `mu` and `sigma` are the mean and standard deviation of the unaltered distribution, respectively.
+where `r` is the raw expression value, and `mu` and `sigma` are the mean and standard deviation of the base population, respectively.
+
+### How to proceed
+cBioPortal expects z-score normalization to take place per gene. You can produce this extra z-score file yourself, or let cBioPortal do it for your input files by using `convertExpressionZscores.pl` (Method 1, diploid samples as base population) OR `NormalizeExpressionLevels.py` (Method 2, all samples as base population). Examples of the calculation and running the programs are below.
```

Comment:
I worry that this wording might confuse a user into thinking that z-scores will be generated as part of the import process. How about rephrasing the second sentence to:
You can calculate z-scores with your own preferred method, or use one of the cBioPortal provided approaches:
- `convertExpressionZscores.pl` applies Method 1 (diploid samples as base population)
- `NormalizeExpressionLevels.py` applies Method 2 (all samples as base population)

Reply:
Hi @tmazor, thank you for the review. This is a good catch. We have updated and reorganized the doc a little bit. Do you mind taking another quick look? Thank you!

Reply:
The reorganization is a great improvement! I did add one new comment about using high/low rather than over/under-expressed, but besides that, this looks great!